json-test.java

```diff
@@ -1,246 +0,0 @@
-                                    var("s2").isa("subject").has("name", "mathematics"),
-                                    var("s3").isa("speciality").has("name", speciality)
-                            ).insert(
-                                    var("new-offer").rel("school", "s").rel("subject", "s2").rel("speciality", "s3").isa("offer")
-                            );
-                            offerSet.add(a);
-                            writeTransaction.query().insert(insertQuery);
-                            writeTransaction.commit();
-                            System.out.println("Offer Relationship added\n"+" school: "+school+" speciality = "+speciality);
-                            System.out.println("size = "+offerSet.size());
-                        }
-                    }
-
-                    //add mastery relationship
-                    try (TypeDBTransaction writeTransaction = session.transaction(TypeDBTransaction.Type.WRITE))
-                    {
-                        TypeQLInsert insertQuery = TypeQL.match(
-                                var("m").isa("mathematician").has("id", m_id),
-                                var("s").isa("speciality").has("name", speciality)
-                        ).insert(
-                                var("new-mastery").rel("master", "m").rel("speciality", "s").isa("mastery")
-                        );
-                        writeTransaction.query().insert(insertQuery);
-                        writeTransaction.commit();
-                    }
-
-                    List advisor_list=get_advisors(file, z);
-                    int len=advisor_list.size();
-
-                    //add research relationship
-                    for (int j=0; j<len; j++)
-                    {
-                        java.lang.Long a_id=(java.lang.Long)advisor_list.get(j);
-                        //research relationship
-                        try (TypeDBTransaction writeTransaction = session.transaction(TypeDBTransaction.Type.WRITE))
-                        {
-                            TypeQLInsert insertQuery = TypeQL.match(
-                                    var("m1").isa("mathematician").has("id", m_id),
-                                    var("m2").isa("mathematician").has("id", a_id),
-                                    var("s").isa("school").has("name", school)
-                            ).insert(
-                                    var("new-research").rel("advisor", "m2").rel("advisee", "m1").rel("school", "s").isa("research")
-                            );
-                            writeTransaction.query().insert(insertQuery);
-                            writeTransaction.commit();
-                        }
-                    }
-                }
-            }
-        }
-        catch(IOException e)
-        {
-            System.out.println("IO Exception in file: "+e.getMessage());
-            e.printStackTrace();
-        }
```

Comment:
Can the IOException be thrown from anywhere in the code? I expect it to be only the line which reads the file.
Try to keep the try-catch block as small as possible. In this case, I'd just return/exit if the file-loading failed.